### ISSUE TITLE
Deleted item values from nested group are added back on adding a new item

### ISSUE THREAD

**Issue by jeevananthank** [Altinn/app-frontend-react#128]
## Describe the bug
When deleting an item from the nested group, and adding a new item in the group, the fields in the nested group is filled with the values from the deleted item

## To Reproduce
1. Start an app instance with nested group configured.
2. Add an item to the main group and a new item to the nested group and fill the value.
3. Delete the nested group item and add a new item to the nested group.
4. the fields in the group is prefilled with the values from the deleted item.

## Expected behavior
The fields on the new item in the nested group should not be filled with value from the deleted item.

## Additional info
env: localtest
app: ttd/frontend-test

**Comment by olemartinorg** [comment]
FYI testers: I found this bug while implementing attachments in repeating groups (#246) and fixed it in that PR. There is also a cypress test where this (erroneous) behavior was tested, and I changed that test when implementing #246.

You can run this to see the change:
`git diff f481331..1ae84e4 -- test/cypress/e2e/integration/app-frontend/group.js`

**Comment by nkylstad** [comment]
@olemartinorg  Tested ok in frontend-test app, new item is not prefilled with values from deleted item.

However I see that it's not possible to completely delete the item if there is only one item in the nested group. The content of the item is deleted, but the item itself is still in the list (although empty). If I then click "add new item", a second item is added.

![image.png](https://images.zenhubusercontent.com/5dc93d9470573a0001558cc6/222fd2ba-ad02-4b09-819f-76a14a64a38d)

![image.png](https://images.zenhubusercontent.com/5dc93d9470573a0001558cc6/64c6fc39-eae4-490a-a63f-35c545d9f28a)

I would expect to be able to completely delete the item, even if it is the only item in the nested group. 
The behaviour works as expected for the parent group. I'm not sure if this was an existing issue for nested groupsor if it has been introduced with #246. I don't see an existing issue for it at least.

**Comment by olemartinorg** [comment]
This is partly caused by the main group having `openByDefault` off, but the subgroup having it on. In addition to that, I actually broke that `openByDefault` functionality in [this commit](https://github.com/Altinn/app-frontend-react/pull/246/commits/e2e4c4bd195bb6da491ce72c38c170a7eb42b85c) where I removed the events dispatched to set the `editIndex`. The old approach is no longer possible - as deleting a group row can now be rejected/cancelled (if deleting attachments inside that group fails). I have a fix somewhat ready, and will write cypress tests for it.
